### ISSUE TITLE
PropertyUsedBeforeDeclaration false positive when property and utility function names match

### ISSUE THREAD

**Issue by brentwatson** [detekt/detekt#9147]
## Observed Behavior

If you import a function with the same name as a var being set using that function, detek reports a `PropertyUsedBeforeDeclaration` violation:

```kotlin
import com.xyz.accountCapabilities // this is a function

class Demo {
   override var accountCapabilities: AccountCapabilities? = accountCapabilities() // detekt reports "PropertyUsedBeforeDeclaration - ['accountCapabilities' is before declaration.]"
}
```

This works (detekt does not report an issue) if you use an import alias to rename the function being imported:

```kotlin
import com.xyz.accountCapabilities as helperFunction

class Demo {
   override var accountCapabilities: AccountCapabilities? = helperFunction() // Ok
}
```

## Your Environment

Version of detekt used: 1.23.8
Version of Gradle used: 9.2.1
Version of Kotlin used: 2.3.10
Operating System and version: MacOS Tahoe 26.1

**Comment by atulgpt** [comment]
Hi @brentwatson I added the TC for your case, and is not able to repro you scenario, can you update your detekt version and try with that?

**Comment by brentwatson** [comment]
Closing. See: https://github.com/detekt/detekt/issues/9145#issuecomment-4085047013
